client/app/components/sections/SearchPage/SearchPage.js

```diff
@@ -22,6 +23,35 @@ class SearchPage extends Component {
     super(props, context);
     this.listings = listingsByIds(props.searchPage.listings, props.searchPage.currentPage) || [];
     this.listingProps = this.listingProps.bind(this);
+    this.nextPage = this.nextPage.bind(this);
+    this.prevPage = this.prevPage.bind(this);
+
+    this.totalPages = Math.ceil(this.props.searchPage.state.get('total') / this.props.searchPage.state.get('per_page'));
+    this.hasNextPage = this.totalPages > this.props.searchPage.state.get('page');
+    this.hasPrevPage = this.props.searchPage.state.get('page') > 1;
+  }
+
+  nextPage() {
+    return this.props.searchPage.state.get('page') + 1;
+  }
+
+  prevPage() {
+    return this.props.searchPage.state.get('page') - 1;
+  }
+
+  setPage(num) {
+    return (e) => {
+      e.preventDefault();
+      const next = this.pageUrl(num);
+      window.location = next;
```

Comment:
```window.location = this.pageUrl(num);```
(const *next* is probably just confusing due to another meaning in this context.)